src/Workflow/Activity/OperationalTask.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+
+namespace PHPMentors\Workflower\Workflow\Activity;
+
+
+use PHPMentors\Workflower\Workflow\Operation\OperationalInterface;
+
+abstract class OperationalTask extends Task implements OperationalInterface
+{
+    /**
+     * @var int|string
+     */
+    protected $operation;
+
+    public function __construct(array $config = [])
+    {
+        parent::__construct($config);
+
+        foreach ($config as $name => $value) {
+            if (property_exists(self::class, $name)) {
+                $this->{$name} = $value;
+            }
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function serialize()
+    {
+        return serialize([
+            get_parent_class($this) => parent::serialize(),
+            'operation' => $this->operation,
+        ]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function unserialize($serialized)
+    {
+        foreach (unserialize($serialized) as $name => $value) {
+            if ($name == get_parent_class($this)) {
+                parent::unserialize($value);
+                continue;
+            }
+
+            if (property_exists($this, $name)) {
+                $this->$name = $value;
+            }
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getOperation()
+    {
+        return $this->operation;
+    }
+
+    /**
+     * @param int|string $operation
+     */
+    public function setOperation($operation): void
+    {
+        $this->operation = $operation;
+    }
+
+    public function createWork(): void
+    {
+        // create work items
+        parent::createWork();
+
+        $workflow = $this->getWorkflow();
+        $operationRunner = $workflow->getOperationRunner();
+
+        $participant = $operationRunner->provideParticipant($this, $workflow);
+
+        // execute work items
+        foreach ($this->workItems as $workItem) {
+            if (!$workItem->isEnded()) {
+                $workItem->allocate($participant);
+                $workItem->start();
+                $operationRunner->runWorkItem($workItem);
```

Comment:
https://github.com/phpmentors-jp/workflower-bundle/blob/master/src/Workflow/Operation/OperationRunnerDelegate.php is an implementor of `OperationRunnerInterface`, its objects execute `run(OperationalInterface, Workflow)` for their actual operations.

- Should `run()` be replaced with `runWorkItem(WorkItemInterface)` ?
- If so, it's time to think about merge these methods. Do you think `run(OperationalInterface, WorkItemInterface)` for this PR?

Reply:
Oh, I misunderstood, `ItemInterface` provides the parent instance and activity, right? If so, then `run(WorkItemInterface)` is better than `run(OperationalInterface, WorkItemInterface)`.

Reply:
Yes, that `run` could be replaced by `runWorkItem`.
For the 2nd question, the workItem already has a reference to its parent operational activity (`workitem->getParentActivity()`) so `run(WorkItemInterface)` should be enough.